src/DataFactory.js

```diff
@@ -141,13 +150,13 @@ class DataFactory {
         })
 
         if (tag === -1) {
-            throw new Error('Unknown variant: ' + JSON.stringify(value))
+            throw new Error('Unknown variant: ' + variantCtor)
         }
 
         const [[, variantTypes]] = Object.entries(type.variants[tag])
         const variantValue = this.create(variantTypes, variantArgs)
```

Comment:
> Please move the implementation into FateVariant, I prefer to keep the DataFactory as general as possible

Should `this.create` be here or do we need to invoke it from FateVariant, solving cycle dependency somehow?

If it will stay be here, should the preparation of its arguments (based on `type` and `value`) be extracted as a static method of FateVariant? I don't have any better ideas in mind

Reply:
Keep it as is, I'd re-factor the datafactory anyway